R/endpoints.R

```diff
@@ -122,6 +123,36 @@ endpoint_model_result <- function(queue) {
   }
 }
 
+endpoint_model_debug <- function(queue) {
+  function(req, res, id) {
+    response <- with_success(queue$queue$task_data(id))
+    if (!response$success) {
+      response$errors <- hintr_errors(list(
+        "INVALID_TASK" = response$message))
+      res$status <- 400
+      return(hintr_response(response, "ModelDebugResponse"))
+    }
+
+    data <- response$value
+    files <- vcapply(data$objects$data, identity, USE.NAMES = FALSE)
+    tmp <- tempfile()
+    path <- file.path(tmp, id)
+    dir.create(path, FALSE, TRUE)
+
+    path_files <- file.path(path, "files")
+    dir.create(path_files)
+    file_copy(files, file.path(path_files, basename(files)))
```

Comment:
We copy the files in as their basename only. In case of duplicate files that's going to be a bit ugly. If all 6 files are not required, then this will fail - can you comment @r-ash?

Reply:
Duplicate files might not necessarily fail validation. Thinking that they could upload the same file for the programme or survey data (as long as all the expected rows from every file are available in what they upload). We don't expect to ever get a duplicate file but they could potentially do it. It would take some effort though.

I don't think we will ever get an occasion where all 6 files are not required. Doing a model run with no data would be pretty meaningless? At least as we're currently working 4 files are required and I'm not expecting that to change.